### PR TITLE
feat(events): application enrollment with form builder

### LINKED ISSUE #237: feat(events): participation module MVP
## Problem Statement

A He4rt Developers organiza meetups presenciais, workshops e conferências (incluindo um meetup em breve e uma conferência de 2 dias no final do ano), mas não possui um sistema próprio para gerenciar o ciclo de participação dos membros. Hoje não há como:
- Controlar inscrições com capacidade limitada e lista de espera
- Verificar presença de quem realmente compareceu (check-in)
- Recompensar participação com XP no sistema de gamificação existente
- Aprovar candidatos a eventos com vagas limitadas (workshops)

## Solution

Construir o módulo de participação em eventos (`app-modules/events/`) com lifecycle completo: enrollment → check-in → attendance → XP reward. O sistema suporta 3 métodos de entrada (RSVP, RSVP+check-in, Application com formulário dinâmico), 3 métodos de check-in (manual, código numérico, QR code), waitlist com promoção FIFO, e integração com o módulo de gamificação via domain events.

## User Stories

1. As a participant, I want to RSVP to a meetup with one click, so that I can confirm my presence quickly
2. As a participant, I want to join a waitlist when an event is full, so that I can be promoted if someone cancels
3. As a participant, I want to be automatically promoted from the waitlist when a slot opens, so that I don't miss the opportunity
4. As a participant, I want to cancel my enrollment before the deadline, so that my slot is freed for others
5. As a participant, I want to check in at a meetup by typing a numeric code, so that my presence is verified
6. As a participant, I want to check in at a conference by showing my QR code badge, so that attendance is tracked per day
7. As a participant, I want to receive XP when I confirm, check in, and complete attendance, so that my participation is rewarded in the gamification system
8. As a participant, I want to fill an application form to apply for a workshop, so that the organizer can evaluate my candidacy
9. As a participant, I want to be notified when my application is approved or rejected, so that I know my status
10. As a participant, I want to check in on each day of a multi-day conference, so that my attendance per day is tracked
11. As a participant, I want to use the same QR code across all event days, so that I only need one badge
12. As a participant, I want to check in via Discord/Twitch bot by typing a command with the code, so that I can confirm presence in online events
13. As a participant, I want to see my enrollment status for each event, so that I know where I stand
14. As an organizer, I want to create events with configurable enrollment policies, so that each event has appropriate participation rules
15. As an organizer, I want to set capacity limits with optional waitlist, so that I can control attendance
16. As an organizer, I want to choose the enrollment method (RSVP, RSVP+check-in, Application), so that the entry flow matches the event type
17. As an organizer, I want to choose the check-in method (manual, numeric code, QR code), so that presence verification fits the event format
18. As an organizer, I want to define a dynamic application form (JSONB schema), so that I can collect relevant information per event
19. As an organizer, I want to approve or reject applications with a reason, so that participants understand the decision
20. As an organizer, I want to generate time-limited numeric codes bound to a specific date, so that check-in is controlled per day
21. As an organizer, I want to manually check in participants, so that I can handle edge cases (late arrivals, code issues)
22. As an organizer, I want to override a participant's status (e.g., no_show → attended), so that I can correct errors
23. As an organizer, I want to set a cancellation deadline in hours before the event, so that I have predictability on attendance
24. As an organizer, I want to configure XP rewards per event (RSVP, check-in, attendance), so that different events reward differently
25. As an organizer, I want to set attendance_requirement (all_days, any_day, minimum_days), so that multi-day events have clear completion criteria
26. As an organizer, I want to see all enrollments with their current status, so that I can manage the event
27. As an organizer, I want a full audit trail of every status transition, so that I can investigate disputes
28. As the system, I want to automatically mark checked_in → attended after the event ends, so that XP is awarded without manual intervention
29. As the system, I want to automatically mark confirmed → no_show after the event ends, so that non-attendees are identified
30. As the system, I want enrollment capacity checks to be atomic (lockForUpdate), so that overbooking is impossible under concurrency

### Architecture

- **Module**: `app-modules/events/` following existing modular architecture (`internachi/modular`)
- **Namespace**: `He4rt\Events\*`
- **Panels**: Admin panel = organizer, App panel = participant
- **Table prefix**: All tables prefixed with `events_` for clear identification in database

### State Machine

- Implemented as a PHP backed enum (`EnrollmentStatusEnum`) with `canTransitionTo()` method
- **NOT** using `spatie/laravel-model-states` — transitions validated in Action classes explicitly
- 8 states: `pending`, `confirmed`, `waitlisted`, `checked_in`, `attended`, `cancelled`, `rejected`, `no_show`
- Terminal states: `attended` (success), `cancelled`, `rejected`, `no_show`
- See ADR-0001

### Schema (7 tables)

**`events`** — Core event record:
- `id`, `tenant_id`, `event_type` (meetup/workshop/conference), `active`, `slug`, `title`, `description`, `location`, `starts_at`, `ends_at`, `timestamps`
- No denormalized counters — derived by query

**`events_enrollment_policies`** — 1:1 configuration per event:
- `event_id` (unique), `enrollment_method`, `check_in_method`, `capacity` (nullable = unlimited), `has_waitlist`, `attendance_requirement` (all_days/any_day/minimum_days), `cancellation_deadline_hours`, `xp_reward_rsvp`, `xp_reward_checkin`, `xp_reward_attendance`, `application_form_schema` (JSONB), `timestamps`

**`events_enrollments`** — Hub central (1 per user+event):
- `id`, `event_id`, `user_id`, `status` (enum), `waitlist_position` (nullable), `application_data` (JSONB), `rejection_reason`, `enrolled_at`, `confirmed_at`, `checked_in_at`, `attended_at`, `cancelled_at`, `is_public`, `timestamps`
- UNIQUE constraint on `(event_id, user_id)`

**`events_enrollment_transitions`** — Append-only audit trail:
- `id`, `enrollment_id`, `from_status`, `to_status`, `triggered_by` (system/user/admin), `actor_user_id` (nullable), `reason` (nullable), `created_at`
- Written explicitly by Actions, NOT by database triggers (ADR-0003)

**`events_check_ins`** — 1:N per enrollment (one per day):
- `id`, `enrollment_id`, `method` (manual/numeric_code/qr_code), `payload` (JSONB), `event_date`, `checked_in_at`, `timestamps`
- Multi-day events: one check-in record per day attended (ADR-0004)
- Event days derived from `events.starts_at`/`ends_at` date range — no separate days table

**`events_check_in_codes`** — Organizer-generated numeric codes:
- `id`, `event_id`, `code`, `event_date`, `valid_from`, `valid_until`, `max_uses` (nullable), `uses_count`, `timestamps`
- Bound to specific date — one code per day

**`events_qr_tokens`** — One per enrollment, reusable across days:
- `id`, `enrollment_id`, `token` (unique), `expires_at` (nullable), `timestamps`

### XP Integration

- Events module publishes domain events: `EnrollmentConfirmed`, `ParticipantCheckedIn`, `ParticipantAttended`
- Gamification module listens and calls `IncrementExperience` on Character
- XP amounts come from `events_enrollment_policies` fields, passed as payload in domain events
- No XP ledger in Events module — auditability via `events_enrollment_transitions`
- Idempotency responsibility on Gamification listener
- See ADR-0002

### Bot Integration

- Discord/Twitch bot dispatches a domain event (e.g., `CheckInRequested`) with user identifier and code
- Events module has a listener that validates the code and processes the check-in
- Bot is pure transport — all business logic in Events module
- Bot must listen for response event (`CheckInProcessed`) to reply to user in chat
- Consistent with existing Bot Discord → Moderation pattern in CONTEXT-MAP
- See ADR-0005

### Enrollment Flow — Concurrency

- Capacity check uses `DB::transaction` with `lockForUpdate()` on the enrollment policy
- Count confirmed/checked_in/attended enrollments → compare against capacity → insert or waitlist
- Waitlist promotion on cancellation: FIFO (ordered by `enrolled_at`), atomic update + notification dispatch

### Scheduled Job

- Runs after `event.ends_at` passes
- Batch transitions: `checked_in → attended` (if attendance_requirement met), `confirmed → no_show`
- Dispatches XP domain events for newly `attended` enrollments
- Single job handles both transitions per event

### Application Form

- Dynamic form schema stored as JSONB in `events_enrollment_policies.application_form_schema`
- Filament form builder for organizer to define questions
- Participant responses stored in `events_enrollments.application_data` as JSONB
- Organizer reviews in Admin panel with approve/reject actions

### Event Types

```php
enum EventTypeEnum: string {
    case Meetup = 'meetup';
    case Workshop = 'workshop';
    case Conference = 'conference';
}
```

## Testing Decisions

- **Good tests**: test external behavior through Actions and domain events, not internal implementation. Assert state transitions, database state, and dispatched events.
- **Test location**: `app-modules/events/tests/Feature/` and `app-modules/events/tests/Unit/`
- **Prior art**: existing tests in `app-modules/moderation/tests/` and `app-modules/bot-discord/tests/`
- **What to test**:
  - Each Action (EnrollUser, CancelEnrollment, CheckIn, ApproveApplication, RejectApplication, PromoteFromWaitlist) — happy path + edge cases
  - State machine transitions — valid and invalid
  - Concurrency: capacity enforcement under race conditions
  - Waitlist promotion FIFO ordering
  - Multi-day check-in with attendance_requirement validation
  - Scheduled job (attended + no_show marking)
  - Domain event dispatch (assert events are fired, don't test Gamification's listener here)
  - Cancellation deadline enforcement
  - Numeric code validation (expiry, max_uses, wrong date)
  - QR token validation (expiry, reuse across days)
- **Factories**: create factories for EventModel, EventEnrollmentPolicy, EventEnrollment, EventCheckIn, EventCheckInCode, EventQrToken
- **Use Pest** with `php artisan make:test --pest`
- **ActingAs**: all Filament panel tests require `$this->actingAs(User::factory()->create())`

## Out of Scope

- Networking between participants (profiles, connections, discovery)
- Referral / invite links with tracking
- Magic link check-in method
- Geolocation check-in method
- Sponsors association with events
- Timeline / activity feed display
- Call for Papers / Submissions (CFP)
- Agenda / schedule display
- Paid events / payment integration
- No-show penalties affecting future eligibility (data tracked, consequences deferred)
- Event days as explicit table (derived from date range)
- Denormalized counters on events table

## Further Notes

- The events module was previously gutted (all tables dropped in migration `2026_05_16_195205`). This PRD rebuilds the module from scratch with the new participation-focused architecture.
- The previous module had: events, events_talks, events_attendees, sponsors, events_sponsors, events_agenda, event_submission_speakers. The new module replaces this with a fundamentally different schema focused on enrollment lifecycle.
- Existing ADRs document key architectural decisions: see `app-modules/events/docs/adr/0001-0005`.
- `CONTEXT-MAP.md` at repo root has been updated to include Events and Gamification bounded contexts with dependency diagram.
- `app-modules/events/CONTEXT.md` contains the full domain glossary and state machine reference.
- Future phases (post-MVP): magic_link/geolocation check-in, CFP/submissions, agenda display, networking, referral, payment integration, no-show penalties.

### LINKED ISSUE #243: feat(events): application enrollment with form builder
## Parent

#237

## What to build

Implement the Application enrollment method: participant fills a dynamic form, enrollment goes to `pending`, organizer approves or rejects in Admin panel.

**Dynamic form schema:**
- `events_enrollment_policies.application_form_schema` stores a JSONB array defining form fields
- Schema structure: `[{type: "text|textarea|select|checkbox", label: "...", required: bool, options?: [...]}]`
- Filament form builder in Admin for organizer to define questions when creating/editing the policy
- App panel renders the form dynamically from the schema using Filament form components

**Modify `EnrollUserAction` (application path):**
- When `enrollment_method = 'application'`:
  - Validates submitted data against the form schema
  - Creates enrollment with status `pending`, stores answers in `application_data` (JSONB)
  - Records transition (from=null, to=pending, triggered_by=user)
  - Does NOT check capacity yet (only checked on approval)

**Action: `ApproveApplicationAction`**
- Validates: enrollment status must be `pending`
- Checks capacity (with `lockForUpdate`) — if full + has_waitlist → `waitlisted`, if full + no waitlist → reject approval
- Transitions to `confirmed` (or `waitlisted`), sets `confirmed_at`
- Records transition (triggered_by=admin, actor_user_id=organizer)
- Dispatches `EnrollmentConfirmed` domain event

**Action: `RejectApplicationAction`**
- Validates: enrollment status must be `pending`
- Transitions to `rejected`, stores `rejection_reason`
- Records transition (triggered_by=admin, actor_user_id=organizer, reason=rejection_reason)

**Admin panel:**
- "Pending Applications" view/filter on enrollments — shows application_data in a readable format
- "Approve" and "Reject" actions (reject requires reason text field)
- Application data displayed in enrollment detail view

**App panel:**
- Dynamic form rendered from policy schema when user clicks "Apply"
- Shows "Application pending" status with submitted data (read-only)
- Shows "Rejected" status with rejection reason when applicable

## Acceptance criteria

- [ ] Organizer can define a dynamic form schema via Filament form builder in Admin
- [ ] Participant sees and fills the dynamic form in App panel
- [ ] Submission creates enrollment with status `pending` and `application_data` populated
- [ ] Organizer can approve → enrollment transitions to `confirmed` (or `waitlisted` if full)
- [ ] Organizer can reject with reason → enrollment transitions to `rejected` with reason stored
- [ ] Capacity is checked on approval, not on submission
- [ ] Application data is displayed in Admin enrollment detail
- [ ] Transition audit trail records approver/rejector identity
- [ ] Feature tests: submit application, approve, reject, approve when full (waitlist), approve when full (no waitlist)
- [ ] Pint passes

## Blocked by

- #240 (RSVP Enrollment — shares EnrollUserAction base)

### PR DESCRIPTION
**Pull request by AdryanneKelly**
## Summary

Implements the Application enrollment method: the participant fills a dynamic form defined by the organizer, the enrollment lands as `pending`, and the organizer approves or rejects it from the Admin panel. Reuses `EnrollUserAction` as the shared entrypoint (RSVP path untouched) and adds two new domain actions for the approval/rejection decision.

Flow: participant submits form → `EnrollUserAction` (pending, no capacity check) → Admin approves/rejects → `ApproveApplicationAction` (capacity check + confirm/waitlist) or `RejectApplicationAction` (reason + rejected)

Closes #237

## What changed — app-modules/events

| File | Role |
|---|---|
| `src/Enrollment/Actions/EnrollUserAction.php` | Application path: creates `pending` enrollment, validates `application_data` against the policy's schema (per-type: `required`, `select`/`checkbox` options), skips capacity check on submission. |
| `src/Enrollment/Actions/ApproveApplicationAction.php` | Validates status is `pending`, capacity check with `lockForUpdate`, transitions to `confirmed`/`waitlisted` (or throws if full with no waitlist), dispatches `EnrollmentConfirmed`/`EnrollmentWaitlisted`, audit trail (`triggered_by=admin`, `actor_id`). |
| `src/Enrollment/Actions/RejectApplicationAction.php` | Validates status is `pending`, stores `rejection_reason`, transitions to `rejected`, audit trail with reason. |
| `src/Enrollment/DTOs/ApproveApplicationDTO.php`, `RejectApplicationDTO.php` | New DTOs for the two admin actions. |
| `src/Enrollment/DTOs/EnrollUserDTO.php` | Adds `applicationData` payload. |
| `src/Enrollment/Exceptions/EnrollmentException.php` | Adds `enrollmentNotPending()` and `applicationDataInvalid()` factories. |
| `src/Enrollment/Models/Enrollment.php`, `EnrollmentPolicy.php` | Corrects `@property` typing for `application_data` / `application_schema` (list, not associative array) — removes a PHPStan type-mismatch cascade in the new validation code. |
| `database/factories/EnrollmentPolicyFactory.php`, `database/seeders/EventsSeeder.php` | `application()` factory state + seed data covering all 4 field types (`text`/`textarea`/`select`/`checkbox`) across two Application-method events. |
| `lang/{en,pt_BR}/{exceptions,pages}.php` | New user-facing strings. |

## What changed — app-modules/panel-admin

| File | Role |
|---|---|
| `Filament/Resources/Events/Schemas/EventForm.php` | Repeater-based form builder for `application_schema` (type/label/required/options), visible only when `enrollment_method = Application`. |
| `Filament/Resources/Events/RelationManagers/Actions/ApproveApplicationAction.php`, `RejectApplicationAction.php` | Table row actions — approve (confirmation modal) / reject (reason textarea) — call into the domain actions. |
| `Filament/Resources/Events/RelationManagers/EnrollmentsRelationManager.php` | Wires the two actions plus a "View Application" modal on pending/decided rows. |
| `resources/views/enrollments/application-data.blade.php` | Read-only rendering of submitted answers (boolean, array, and scalar aware). |

## What changed — app-modules/panel-app

| File | Role |
|---|---|
| `Livewire/Events/EventDetail.php` | `canApply` computed prop, `apply()` action, pre-initializes checkbox-group state as arrays in `mount()`. |
| `resources/views/livewire/events/event-detail.blade.php` | Dynamic form rendered per field type; `pending`/`rejected` status views showing the submitted answers / rejection reason. |

## Design notes (the non-obvious bits)

**`checkbox` = multi-select group, not a single boolean.** The issue's schema (`{type, label, required, options?}`) doesn't spell this out, so it was ambiguous whether `checkbox` meant a single yes/no toggle or a group of checkboxes sharing one `options` list (like `select`, but multi-value). Went with the multi-select reading — `checkbox` now shows the same `options` builder as `select` in the Admin, and stores an array of selected values. Validation treats `required` as "at least one selected" and checks each selection against `options`.

**Livewire needs the array pre-seeded.** Multiple `<input type="checkbox" wire:model="applicationFormData.{index}" value="...">` bound to the same array path only get grouped into an array by Livewire's front end if that path is *already* an array before the first interaction — otherwise every checkbox in the group ends up sharing one boolean and checking one checks them all. `EventDetail::mount()` now seeds `applicationFormData[$index] = []` for every `checkbox` field up front.

**Filament enum-cast gotcha.** `Select::make('enrollment_method')->options(EnrollmentMethod::class)` auto-applies an `EnumStateCast`, so `$get('enrollment_method')` inside `visible()` closures returns the enum *instance*, not `->value`. The Repeater's visibility check was comparing against the string value and silently never showed — fixed by comparing against the enum case directly.

## Acceptance criteria

- [x] Organizer can define a dynamic form schema via Filament form builder in Admin
- [x] Participant sees and fills the dynamic form in App panel
- [x] Submission creates enrollment with status `pending` and `application_data` populated
- [x] Organizer can approve → enrollment transitions to `confirmed` (or `waitlisted` if full)
- [x] Organizer can reject with reason → enrollment transitions to `rejected` with reason stored
- [x] Capacity is checked on approval, not on submission
- [x] Application data is displayed in Admin enrollment detail
- [x] Transition audit trail records approver/rejector identity
- [x] Feature tests: submit application, approve, reject, approve when full (waitlist), approve when full (no waitlist)
- [x] Pint passes

## Test plan

```
# Application enrollment tests
vendor/bin/pest app-modules/events/tests/Feature/EnrollUserActionTest.php \
  app-modules/events/tests/Feature/Enrollment/ApproveApplicationActionTest.php \
  app-modules/events/tests/Feature/Enrollment/RejectApplicationActionTest.php \
  app-modules/panel-app/tests/Feature/Events/ApplicationEnrollmentTest.php

# Full events + panel-app suite (must stay green)
vendor/bin/pest app-modules/events/tests app-modules/panel-app/tests

# Style
vendor/bin/pint --test app-modules/events app-modules/panel-admin app-modules/panel-app
```

Results:
- Application enrollment tests: 34/34
- events + panel-app suite: 181/181 (0 regressions)
- Pint: clean

Scenarios covered: submit → pending + audit trail · missing `applicationData` → exception · missing required field → exception · no schema defined → pending · event past → rejected · approve → confirmed + audit trail · approve at capacity + waitlist → waitlisted · approve at capacity, no waitlist → event full exception · approve non-pending → exception · no capacity limit → always confirms · reject with reason → rejected · reject non-pending → exception · reject preserves `application_data` · apply button rendering · `canApply`/`canConfirmPresence` flags · pending status shows submitted answers · rejected shows reason · already applied → no apply button · missing required field → error notification.

**Note on PHPStan:** the module already carried ~172 pre-existing errors before this branch (untyped test-helper array params, Pest fluent-call false positives from Larastan) — out of scope here. This PR's own new type mismatch (`EnrollUserAction::validateApplicationData` against the `application_schema`/`application_data` docblocks) was fixed; remaining new findings are the same pre-existing categories, consistent with every other test file in the suite.

## Out of scope (separate issues)

- Admin bulk actions on pending applications (filter/view exist; bulk approve/reject not implemented).
- Editing a submitted application before it's reviewed.

### CODE REVIEW

File: app-modules/panel-app/src/Livewire/Events/EventDetail.php [hunk 1]
```diff
@@ -155,6 +165,24 @@ public function canConfirmPresence(): bool
...
     }
 
+    #[Computed]
+    public function canApply(): bool
+    {
+        if ($this->event->status !== EventStatus::Published) {
+            return false;
+        }
+
+        if ($this->enrollment !== null) {
```
Comment (BrunaDomingues) [inline app-modules/panel-app/src/Livewire/Events/EventDetail.php]: Não sei se seria interessante deixar mais explicito esse if. Se for permitir reapply, sugiro tratar explicitamente os status que bloqueiam nova candidatura (ex.: pending, confirmed, waitlisted) em vez de “qualquer enrollment existente”.
 if ($this->enrollment !== null && ! $this->enrollment->status->isRejected()) {
    return false;
}

**Review by GabrielFVDev — APPROVED** [review]
Image

Se der para ajustar isso, ao inves de trazer o id do usuario, ser o nome dele.
Fica show!

De resto, LGTM

**Review by BrunaDomingues — COMMENTED** [review]
<html>
<body>
<h3 data-pm-slice="1 1 []">Importante — strings hardcoded em inglês (i18n)</h3><p><strong>Onde:</strong></p>
image

<p><strong>Por quê:</strong> módulo <code>events</code> já tem <code>lang/en</code> e <code>lang/pt_BR</code> para pages/exceptions; Admin/App ficam inconsistentes.</p><p><strong>Sugestão:</strong> mover para <code>events::pages.*</code> ou chaves Filament traduzíveis (<code>__()</code>), como no resto do módulo.</p>
</body>
</html>

**Review by BrunaDomingues — APPROVED** [review]
Só deixei o comentário das traduções mas de resto tá top!

**Review by davicbtoliveira — APPROVED** [review]
LGTM

File: app-modules/panel-admin/src/Filament/Resources/Events/Schemas/EventForm.php [hunk 2]
```diff
@@ -153,13 +154,38 @@ public static function configure(Schema $schema): Schema
                             ->minValue(0)
                             ->default(0),
 
-                        KeyValue::make('application_schema')
-                            ->label('Application Form Schema')
-                            ->keyLabel('Field name')
-                            ->valueLabel('Field type / label')
+                        Repeater::make('application_schema')
```
Comment (hefeus) [inline app-modules/panel-admin/src/Filament/Resources/Events/Schemas/EventForm.php:157]: As respostas ficam amarradas na posição da pergunta no schema. Como o Repeater deixa reordenar e remover perguntas, se o organizador mexer no form depois de já ter candidatura, as respostas vão aparecer embaixo da pergunta errada no View Application — e ninguém vai perceber que tá errado.

Acho que vale cada pergunta ter um id fixo e as respostas apontarem pra ele. Ou, se não quiser mexer nisso agora, bloquear edição do form quando já tiver candidatura.

**Review by danielhe4rt — APPROVED** [review]
